graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -189,22 +192,27 @@ def __init__(self, renamings, query_type, scalar_types):
         self.builtin_types = frozenset({'String', 'Int', 'Float', 'Boolean', 'ID'})
 
     def _rename_name_and_add_to_record(self, node):
-        """Rename the value of the node, and add the name pair to reverse_name_map.
+        """Return a copy of input node with a new name, and add the name pair to reverse_name_map.
 
         Don't rename if the type is the query type, a scalar type, or a builtin type.
 
-        Modifies node and potentially modifies reverse_name_map.
+        The input node will not be modified. reverse_name_map may be modified.
 
         Args:
-            node: type Name (see graphql/language/ast), an object describing the name of an AST
-                  element such as type, interface, or scalar (user defined or builtin)
+            node: EnumTypeDefinition, InterfaceTypeDefinition, NamedTuple, ObjectTypeDefinition,
+                  or UnionTypeDefinition. An object representing an AST component, containing a
+                  .name attribute corresponding to an AST node of type Name.
+
+        Returns:
+            None, if the name was not changed; a Node object that is nearly identical to the
+            input node but with a new name, if the name was changed
 
         Raises:
             - InvalidTypeNameError if either the node's current name or renamed name is invalid
             - SchemaNameConflictError if the newly renamed node causes name conflicts with
               existing types, scalars, or builtin types
         """
-        name_string = node.value
+        name_string = node.name.value
 
         if name_string == self.query_type or name_string in self.scalar_types:
             return
```

Comment:
Should I write `return None` to make it more clear that returning None means no action?

Reply:
I would, yes -- being explicit is always better than relying on implicit behavior, since it's more obvious what is going on. Although I think this is actually not going to matter in the end, since the discussion we had above means that the function contract will change.